crates/starknet-server/src/api/json_rpc/mod.rs

```diff
@@ -0,0 +1,461 @@
+        }
+    }
+}
+
+impl<T: Serialize> ToRpcResponseResult for RpcResult<T> {
+    fn to_rpc_result(self) -> ResponseResult {
+        match self {
+            Ok(data) => to_rpc_result(data),
+            Err(err) => match err {
+                ApiError::RpcError(rpc_error) => rpc_error,
+                err @ ApiError::BlockNotFound => RpcError {
+                    code: server::rpc_core::error::ErrorCode::ServerError(24),
+                    message: err.to_string().into(),
+                    data: None,
+                },
+                err @ ApiError::ContractNotFound => RpcError {
+                    code: server::rpc_core::error::ErrorCode::ServerError(20),
+                    message: err.to_string().into(),
+                    data: None,
+                },
+                err @ ApiError::TransactionNotFound => RpcError {
+                    code: server::rpc_core::error::ErrorCode::ServerError(25),
+                    message: err.to_string().into(),
+                    data: None,
+                },
+                err @ ApiError::InvalidTransactionIndexInBlock => RpcError {
+                    code: server::rpc_core::error::ErrorCode::ServerError(27),
+                    message: err.to_string().into(),
+                    data: None,
+                },
+                err @ ApiError::ClassHashNotFound => RpcError {
+                    code: server::rpc_core::error::ErrorCode::ServerError(28),
+                    message: err.to_string().into(),
+                    data: None,
+                },
+                err @ ApiError::ContractError => RpcError {
+                    code: server::rpc_core::error::ErrorCode::ServerError(40),
+                    message: err.to_string().into(),
+                    data: None,
+                },
+                err @ ApiError::NoBlocks => RpcError {
+                    code: server::rpc_core::error::ErrorCode::ServerError(32),
+                    message: err.to_string().into(),
+                    data: None,
+                },
+                err @ ApiError::RequestPageSizeTooBig => RpcError {
+                    code: server::rpc_core::error::ErrorCode::ServerError(31),
+                    message: err.to_string().into(),
+                    data: None,
+                },
+                err @ ApiError::InvalidContinuationToken => RpcError {
+                    code: server::rpc_core::error::ErrorCode::ServerError(33),
+                    message: err.to_string().into(),
+                    data: None,
+                },
+                err @ ApiError::TooManyKeysInFilter => RpcError {
+                    code: server::rpc_core::error::ErrorCode::ServerError(34),
+                    message: err.to_string().into(),
+                    data: None,
+                },
+            }
+            .into(),
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl RpcHandler for JsonRpcHandler {
```

Comment:
I find this naming confusing: what is RpcHandler and what is JsonRpcHandler, should there ever be some other kind of RpcHandler? I really think something needs to be changed here.

Reply:
I am open for discussion, but the RpcHandler is a trait and JsonRpcHandler is the implementation. Also worse case will be both to have the same names. Anyway this part is inspired from anvil from foundry-rs

Reply:
Sorry, I must have misunderstood in my initial reading and had it the other way around with what is the trait. Resolving.